ccdproc/core.py

```diff
@@ -818,18 +818,25 @@ def flat_correct(ccd, flat, min_value=None, norm_value=None):
 
     # If a norm_value was input and is positive, use it to scale the flat
     if norm_value is not None and norm_value > 0:
-        flat_mean_val = norm_value
+        flat_mean = (
+            norm_value if hasattr(norm_value, "unit") else norm_value * use_flat.unit
+        )
     elif norm_value is not None:
         # norm_value was set to a bad value
         raise ValueError("norm_value must be greater than zero.")
     else:
         # norm_value was not set, use mean of the image.
-        flat_mean_val = use_flat.data.mean()
+        flat_mean = use_flat.mean()
```

Comment:
```suggestion
        flat_mean = use_flat.data.mean()
```

Reply:
Yeah, I'm seeing where this fails for some versions.  It's important that `flat_mean` have the unit given the other change I made below.  If we want to keep the old dependencies, I think I should change this to:

`flat_mean = use_flat.data.mean() * use_flat.unit`

What do you think?

Reply:
Good point, I had forgotten about the unit. Your proposal seems like the right fix